### PR TITLE
eddie work on DSDEEPB-1490

### PR DESCRIPTION
**Pull request by eddiebroad**
add sorting on the status and name  columns

clean up and use capitalization func

### CODE REVIEW

**Comment by MattPutnam** [comment]
I'm getting this:

screen shot 2015-11-02 at 1 48 21 pm

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspaces_list.cljs [hunk 1]
```diff
@@ -108,17 +116,24 @@
...
                                [:div {:style {:padding "1.1em 0 0 14px"
                                               :fontStyle (when-not description "oblique")}}
-                                (or description "No description provided")])}]
+                                (or description "No description provided")])}
+          {:header "Access Level" :starting-width 400 :sort-by WSAccessLevelKeyFunc
+           :sort-initial :asc
+           :content-renderer
+           (fn [accessLevel]
+             [:div {:style {:padding "1.1em 0 0 14px"
+                            :fontStyle (when-not accessLevel "oblique")}}
```
Comment (MattPutnam) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspaces_list.cljs]: Why is this font style here?  If there's no `accessLevel` there won't be any text to see.

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspaces_list.cljs [hunk 2]
```diff
@@ -108,17 +116,24 @@
                    {:text "Running" :pred #(= "Running" (:status %))}
                    {:text "Exception" :pred #(= "Exception" (:status %))}]
          :columns
-         [{:header [:div {:style {:marginLeft -6}} "Status"] :starting-width 60
-           :content-renderer (fn [data] [StatusCell {:data data}])
-           :filter-by :none}
-          {:header "Workspace" :starting-width 450
-           :content-renderer (fn [data] [WorkspaceCell {:data data}])
-           :filter-by :name}
+         [{:sort-by (fn [m] (:status m))
```
Comment (MattPutnam) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspaces_list.cljs]: You can just do `:sort-by :status`.  But, I don't think we should have sorting on this column anyway.

**Comment by eddiebroad** [comment]
Re: "I'm getting this:" wow!  when I loaded it I didn't get that at all.  Does it arrange better if you resize your browser?  I didn't experiment with that

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspaces_list.cljs [hunk 3]
```diff
@@ -108,17 +116,24 @@
...
+           :header [:div {:style {:marginLeft -6}} "Status"] :starting-width 60
+           :content-renderer (fn [data] [StatusCell {:data data}]) :filter-by :none}
+          {:sort-by (fn [m] (:name m))
```
Comment (MattPutnam) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspaces_list.cljs]: `:sort-by :name`.

You can also add `:as-text :name`, then use `:sort-by :text` and get rid of the `:filter-by`.

**Comment by MattPutnam** [comment]
Looks like the browser window needs to be at least 1423 pixels wide in order to render properly, which I only have on my secondary monitor.

**Comment by MattPutnam** [comment]
If you change line 98 of workspaces_list.cljs to:

    :header-style {:padding "0.5em 0 0.5em 14px" :overflow nil}

That will fix the issue with the alignment of the sort arrows.

**Comment by eddiebroad** [comment]
referring to the issue of which you posted a screenshot, some tweaks from commits and PR suggestions helped some.....but on my screen if I resize the window small enough, then the table rows again show up like what you show in the screenshot.  What do you think?

**Comment by MattPutnam** [comment]
Yeah, there's something wrong with the div width calculations that's causing the table rows to wrap.  That's beyond the scope of this issue, so as long as it still looks okay at a standard resolution let's leave it like that for now.

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspaces_list.cljs [hunk 4]
```diff
@@ -87,8 +95,8 @@
          :cell-padding-left nil
          :header-row-style {:fontWeight nil :fontSize "90%"
                             :color (:text-light style/colors) :backgroundColor nil}
-         :header-style {:padding "0 0 1em 14px" :overflow nil}
-         :resizable-columns? false :reorderable-columns? false :sortable-columns? false
+         :header-style {:padding "0.5em 0 0.5em 14px" :overflow nil}
+         :resizable-columns? false :reorderable-columns? false :sortable-columns? true
```
Comment (MattPutnam) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspaces_list.cljs]: sortable columns is the default so you can actually take that property off

**Comment by eddiebroad** [comment]
Re: "Yeah, there's something wrong...." okay.

**Comment by eddiebroad** [comment]
Re: "sortable columns is the default.....".  I will remove the key/val pair

**Pull request merged**